pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/metadata/ColumnMetadataImpl.java

```diff
@@ -279,8 +284,26 @@ public static ColumnMetadataImpl fromPropertiesConfiguration(String column, Prop
     String partitionFunctionName = config.getString(Column.getKeyFor(column, Column.PARTITION_FUNCTION), null);
     if (partitionFunctionName != null) {
       int numPartitions = config.getInt(Column.getKeyFor(column, Column.NUM_PARTITIONS));
+      Map<String, String> partitionFunctionConfigMap = null;
+      Configuration partitionFunctionConfig = config.subset(Column.getKeyFor(column, Column.PARTITION_FUNCTION_CONFIG));
+      if (!partitionFunctionConfig.isEmpty()) {
+        partitionFunctionConfigMap = new HashMap<>();
+        Iterator<String> partitionFunctionConfigKeysIter = partitionFunctionConfig.getKeys();
+        while (partitionFunctionConfigKeysIter.hasNext()) {
+          String functionConfigKey = partitionFunctionConfigKeysIter.next();
+          Object functionConfigValueObj = partitionFunctionConfig.getProperty(functionConfigKey);
+          String valueStr;
+          if (functionConfigValueObj instanceof List) {
```

Comment:
When `,` is used as column value separator, the value read from properties is of type List, hence this handling.

Reply:
Let's add some comment here to describe this

Reply:
done.